app/services/process_write_off_service.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class ProcessWriteOffService < WasteCarriersEngine::BaseService
+  def run(finance_details:, user:, comment:)
+    @finance_details = finance_details
+    @user = user
+
+    return false if finance_details.balance.zero?
+
+    finance_details.payments << build_write_off(comment)
+
+    finance_details.update_balance
+    finance_details.save!
+  end
+
+  private
+
+  attr_reader :user, :finance_details
+
+  def build_write_off(comment)
+    write_off = WasteCarriersEngine::Payment.new(payment_type: payment_type)
+
+    write_off.order_key = SecureRandom.uuid.split("-").last
```

Comment:
What do we do with this `order_key`?

Reply:
Nothing at all. But the old system generates it like that when doing write off, so for now, I have added it as per old system